File: packages/js-sdk/src/regex/detectors/us/tax-id.ts
// US Tax ID / EIN detector (context-required)

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'

class TaxIdDetector extends RegexDetector {
  entityType = EntityType.TAX_ID
  score = 0.80
  contextKeywords = [
    'ein', 'tax id', 'tax identification', 'employer identification',
    'tin', 'tax #', 'ein#',
  ]
  contextRequired = true
  contextWindow = 50

  // EIN format: XX-XXXXXXX (first two digits are valid prefixes)
  pattern = /\b\d{2}-\d{7}\b/g
}

registerRegion('us', TaxIdDetector)

export { TaxIdDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/us/tax_id.py
"""US Tax ID / EIN detector (context-required)"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region


@register_region("us")
class TaxIdDetector(RegexDetector):
    entity_type = EntityType.TAX_ID
    score = 0.80
    context_keywords = [
        "ein", "tax id", "tax identification", "employer identification",
        "tin", "tax #", "ein#",
    ]
    context_required = True
    context_window = 50

    # EIN format: XX-XXXXXXX (first two digits are valid prefixes)
    pattern = re.compile(r"\b\d{2}-\d{7}\b")
